File: Patrones/Comportamiento/Observer/Python/Cliente.py
from Observador import Observador

# Cliente concreto (Observador concreto)
class Cliente(Observador):
    def __init__(self, nombre: str):
        self.__nombre = nombre

    def actualizar(self, producto: str, precio: float):
        print(f"El cliente {self.__nombre} es notificado: El precio del producto {producto} se actualizó a ${precio:.2f}")

File: Patrones/Comportamiento/Observer/Java/src/Cliente.java
// Esta clase es el ConcreteSubscriber
// Es la clase que implementa la interfaz Observador
public class Cliente implements Observador {

    private String nombre;

    public Cliente(String nombre) {
        this.nombre = nombre;
    }

    @Override
    public void actualizar(String producto, double precio) {
        System.out.println("El cliente " + nombre + " es notificado: El precio del producto " + producto + " se actualizo a $"
                + precio);
    }
}